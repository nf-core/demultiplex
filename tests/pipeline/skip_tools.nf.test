nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "skip_trimming"
    tag "pipeline"


    test("Skip Trimming") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                trim_fastq    = false
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions_skip_trimming") },
                { assert workflow.trace.succeeded().size() == 5 },
                { assert path("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz.md5"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("skip_trimming") }
            )
        }

    }

    test("Skip Fastp") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastp"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions_skip_fastp") },
                { assert workflow.trace.succeeded().size() == 4 },
                { assert path("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz.md5"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("skip_fastp") }
            )
        }

    }

    test("Skip Fastqc") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastqc"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions_skip_fastqc") },
                { assert workflow.trace.succeeded().size() == 5 },
                { assert path("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001.fastp.fastq.gz.md5"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("skip_fastqc") }
            )
        }

    }

    test("Skip Fastp & Fastqc") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastp,fastqc"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions_skip_fastp_fastqc") },
                { assert workflow.trace.succeeded().size() == 4 },
                { assert path("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz.md5"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("skip_fastp_fastqc") }
            )
        }

    }

    test("Skip MultiQC") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "multiqc"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions_skip_multiqc") },
                { assert workflow.trace.succeeded().size() == 4 },
                { assert !path("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert snapshot(
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Sample1_S1_L001.fastp.fastq.gz.md5"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/Undetermined_S0_L001_R1_001.fastq.gz"),
                        path("$outputDir/220422_M11111_0222_000000000-K9H97/L001/Reports/").list(),
                    ).match("skip_multiqc") }
            )
        }

    }

}
