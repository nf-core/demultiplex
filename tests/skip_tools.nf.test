nextflow_pipeline {

    name "Test Workflow main.nf - Skip Tools"
    script "../main.nf"
    profile "test"
    tag "skip_trimming"
    tag "pipeline"


    test("Skip Trimming") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                trim_fastq    = false
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() },
                { assert snapshot(UTILS.validateFastqPaths("$outputDir/samplesheet/atacseq_samplesheet.csv")).match("atacseq_samplesheet") },
                { assert snapshot(UTILS.validateFastqPaths("$outputDir/samplesheet/rnaseq_samplesheet.csv")).match("rnaseq_samplesheet") },
                { assert snapshot(UTILS.validateFastqPaths("$outputDir/samplesheet/taxprofiler_samplesheet.csv")).match("taxprofiler_samplesheet") }
            )
        }
    }

    test("Skip Fastp") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastp,samshee"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Skip Fastqc") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastqc,samshee"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Skip Fastp & Fastqc") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "fastp,fastqc,samshee"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Skip MultiQC") {

        when {
            params {
                input         = 'https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/flowcell_input.csv'
                demultiplexer = 'bclconvert'
                outdir        = "$outputDir"
                skip_tools    = "multiqc,samshee"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
