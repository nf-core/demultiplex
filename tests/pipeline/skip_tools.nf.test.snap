{
    "software_versions_skip_trimming": {
        "content": [
            "{BCLCONVERT={bclconvert=00.000.000.4.0.3}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "skip_fastqc": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001.fastp.fastq.gz.md5:md5,c3cd96f3a22fb6afbaa1df324814d54c",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,2df2e405991814571c021dc8749c2a89",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "skip_fastp": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001_R1_001.fastq.gz.md5:md5,c547a3c03e7c60c871d78018ec5d933f",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,2df2e405991814571c021dc8749c2a89",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "skip_multiqc": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001.fastp.fastq.gz.md5:md5,c3cd96f3a22fb6afbaa1df324814d54c",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,2df2e405991814571c021dc8749c2a89",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "skip_fastp_fastqc": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001_R1_001.fastq.gz.md5:md5,c547a3c03e7c60c871d78018ec5d933f",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,2df2e405991814571c021dc8749c2a89",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "software_versions_skip_fastqc": {
        "content": [
            "{BCLCONVERT={bclconvert=00.000.000.4.0.3}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "software_versions_skip_fastp_fastqc": {
        "content": [
            "{BCLCONVERT={bclconvert=00.000.000.4.0.3}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "software_versions_skip_multiqc": {
        "content": [
            "{BCLCONVERT={bclconvert=00.000.000.4.0.3}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, FASTP={fastp=0.23.4}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "software_versions_skip_fastp": {
        "content": [
            "{BCLCONVERT={bclconvert=00.000.000.4.0.3}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FALCO={falco=1.2.1}, Workflow={nf-core/demultiplex=1.4.0dev}}"
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    },
    "skip_trimming": {
        "content": [
            "Sample1_S1_L001_R1_001.fastq.gz:md5,0a0341e2990b4fa1d9ad4b4c603144c1",
            "Sample1_S1_L001_R1_001.fastq.gz.md5:md5,c547a3c03e7c60c871d78018ec5d933f",
            "Undetermined_S0_L001_R1_001.fastq.gz:md5,febef808ae5397ea4ee7ee40e5fd39e0",
            [
                "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                "SampleSheet.csv:md5,2df2e405991814571c021dc8749c2a89",
                "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                "fastq_list.csv:md5,05bc84f51840f5754cfb8381b36f2cb0"
            ]
        ],
        "timestamp": "2023-10-17T08:05:28+0000"
    }
}